pkg/deploy/registry/rest.go

```diff
@@ -28,29 +29,27 @@ func WaitForRunningDeployment(rn kclient.ReplicationControllersNamespacer, obser
 		return observed, false, err
 	}
 	defer w.Stop()
-
 	ch := w.ResultChan()
-	// Passing time.After like this (vs receiving directly in a select) will trigger the channel
-	// and the timeout will have full effect here.
-	expire := time.After(timeout)
-	for {
-		select {
-		case event := <-ch:
-			obj, ok := event.Object.(*kapi.ReplicationController)
-			if !ok {
-				return observed, false, errors.New("received unknown object while watching for deployments")
-			}
-			observed = obj
-
-			switch deployutil.DeploymentStatusFor(observed) {
-			case api.DeploymentStatusRunning, api.DeploymentStatusFailed, api.DeploymentStatusComplete:
-				return observed, true, nil
-			case api.DeploymentStatusNew, api.DeploymentStatusPending:
-			default:
-				return observed, false, ErrUnknownDeploymentPhase
-			}
-		case <-expire:
-			return observed, false, nil
+	stopChan := make(chan struct{})
+	isRunning := false
+	wait.Until(func() {
```

Comment:
watch.Until is different from wait.Until: https://github.com/kubernetes/kubernetes/blob/53e9832ba36630f34bd920469ef7983a2488c3e9/pkg/watch/until.go#L35 :)